triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -817,10 +838,11 @@ object Server {
     val serverBinding = for {
       _ <- initializeF
       _ <- Future.traverse(initialDars)(server.addDar(_))
+      metricsInterceptor <- rateDurationSizeMetrics
       binding <- Http()
         .newServerAt(host, port)
         .withSettings(ServerSettings(untypedSystem).withTransparentHeadRequests(true))
-        .bind(server.route)
+        .bind(metricsInterceptor apply server.route)
```

Comment:
Can you add a test to check if this works, please?

Reply:
There is no real straightforward way to test that the metrics gathering is correctly activated.
There are a few levels of indirection between the creation of the traffic service server, and the creation of the metrics elements.

The main problem is that the metrics cannot be created in `ServerMain` before creating the `Server`, because they need the execution context, and it only becomes available when the `Server` is initialized.
The metrics cannot be injected later, because they are needed during the initialization of the `Server`.

I think it could be tweaked, but the `Server` actor would not be the top actor anymore. Not sure what would be the impact.

Reply:
It would make sense to add end-to-end integration tests verifying observability for the trigger-service then — let's address that in a separate PR.